packages/joy-media/src/Upload.tsx

```diff
@@ -97,12 +97,14 @@ class Component extends React.PureComponent<Props, State> {
   }
 
   private renderUploading () {
-    const { file, newContentId } = this.state;
+    const { file, newContentId, progress, error } = this.state;
     if (!file) return <em>Loading...</em>;
 
+    const success = !error && progress >= 100;
+
     return <div style={{ width: '100%' }}>
       {this.renderProgress()}
-      <EditMeta contentId={newContentId} fileName={fileNameWoExt(file.name)} />
+      { success ? <EditMeta contentId={newContentId} fileName={fileNameWoExt(file.name)} /> : null }
```

Comment:
There is a short cut for this with `&&`:
```suggestion
      {success && <EditMeta contentId={newContentId} fileName={fileNameWoExt(file.name)} />}
```